test/plugin/base_test.rb

```diff
@@ -1249,41 +1251,70 @@ def test_log_entry_http_request_field_from_record
     verify_subfields_from_record(DEFAULT_HTTP_REQUEST_KEY)
   end
 
-  def test_log_entry_source_location_field_from_record
-    verify_subfields_from_record(DEFAULT_SOURCE_LOCATION_KEY)
+  def test_log_entry_labels_field_from_record
+    verify_subfields_from_record(DEFAULT_LABELS_KEY, false)
   end
 
   def test_log_entry_operation_field_from_record
     verify_subfields_from_record(DEFAULT_OPERATION_KEY)
   end
 
+  def test_log_entry_source_location_field_from_record
+    verify_subfields_from_record(DEFAULT_SOURCE_LOCATION_KEY)
+  end
+
   # Verify the subfields extraction of LogEntry fields when there are other
   # fields.
 
   def test_log_entry_http_request_field_partial_from_record
     verify_subfields_partial_from_record(DEFAULT_HTTP_REQUEST_KEY)
   end
 
-  def test_log_entry_source_location_field_partial_from_record
-    verify_subfields_partial_from_record(DEFAULT_SOURCE_LOCATION_KEY)
-  end
+  # We don't need a test like 'test_log_entry_labels_field_partial_from_record'
+  # because labels are free range strings. Everything in the labels field should
+  # be in the resulting logEntry->labels field. There is no need to check
+  # partial transformation (aka, some 'labels' fields are extracted, while
+  # others are left as it is).
 
   def test_log_entry_operation_field_partial_from_record
     verify_subfields_partial_from_record(DEFAULT_OPERATION_KEY)
   end
 
+  def test_log_entry_source_location_field_partial_from_record
+    verify_subfields_partial_from_record(DEFAULT_SOURCE_LOCATION_KEY)
+  end
+
   # Verify the subfields extraction of LogEntry fields when they are not hashes.
 
   def test_log_entry_http_request_field_when_not_hash
-    verify_subfields_when_not_hash(DEFAULT_HTTP_REQUEST_KEY)
+    # Leave the malformed field as it is.
+    # TODO(qingling128) On the next major after 0.7.4, make the rest of
+    # logEntry subfields behave the same way: if the field is not a hash as
+    # expected, log an error in the Fluentd log and remove this field from
+    # payload. This is the preferred behavior per PM decision.
+    verify_subfields_untouched_when_not_hash(DEFAULT_HTTP_REQUEST_KEY)
   end
 
-  def test_log_entry_source_location_field_when_not_hash
-    verify_subfields_when_not_hash(DEFAULT_SOURCE_LOCATION_KEY)
+  def test_log_entry_labels_field_when_not_hash
+    verify_subfields_removed_when_not_hash(DEFAULT_LABELS_KEY)
   end
 
   def test_log_entry_operation_field_when_not_hash
-    verify_subfields_when_not_hash(DEFAULT_OPERATION_KEY)
+    # Leave the malformed field as it is.
```

Comment:
This one-line comment probably belongs within `verify_subfields_untouched_when_not_hash`, similarly to `verify_subfields_removed_when_not_hash`.

Reply:
Done